routers/repo/pull.go

```diff
@@ -874,6 +906,38 @@ func MergePullRequest(ctx *context.Context, form auth.MergePullRequestForm) {
 	ctx.Redirect(ctx.Repo.RepoLink + "/pulls/" + com.ToStr(pr.Index))
 }
 
+// CancelAutoMergePullRequest cancels a scheduled pr
+func CancelAutoMergePullRequest(ctx *context.Context) {
+	issue := checkPullInfo(ctx)
```

Comment:
maybe need permission check.

Reply:
Also suggest adding api to do CancelAutoMergePullRequest work, Thanks

Reply:
I think adding `CancelAutoMergePullRequest` to the api would only make sense if it would be possible to schedule a PR to merge when checks succeed through the api. Since that is currently not possible, I'd like to postpone both to another PR after this one is merged to get the nits here resolved and not block this any longer.